tests/test_api.py

```diff
@@ -199,3 +172,30 @@ def test_pytket_with_wasm(self) -> None:
             "args": [],
             "returns": ["c2"],
         }
+
+    def test_pytket_to_phir_no_machine(self) -> None:
+        """Test case when no machine is present."""
+        circuit = get_qasm_as_circuit(QasmFile.baby)
+        assert pytket_to_phir(circuit)
+
+    @pytest.mark.parametrize("test_file", list(QasmFile))
+    def test_pytket_to_phir_no_machine_all(self, test_file: QasmFile) -> None:
+        """Test case when no machine is present."""
+        circuit = get_qasm_as_circuit(test_file)
+
+        match test_file:
+            case QasmFile.big_gate:
+                with pytest.raises(KeyError, match=r".*CnX.*"):
+                    assert pytket_to_phir(circuit)
+            case QasmFile.qv20_0:
+                with pytest.raises(KeyError, match=r".*U3.*"):
+                    assert pytket_to_phir(circuit)
+            case _:
+                assert pytket_to_phir(circuit)
+
+    @pytest.mark.parametrize("test_file", list(QasmFile))
+    def test_pytket_to_phir_h1_1_all(self, test_file: QasmFile) -> None:
+        """Standard case."""
+        circuit = get_qasm_as_circuit(test_file)
+
+        assert pytket_to_phir(circuit, QtmMachine.H1_1)
```

Comment:
This reordering of tests should be unnecessary. I am taking a look why tests fail without it, but it clearly indicated we might be unintentionally storing state somewhere where we should not be.

Reply:
I couldn't find the root cause of the issue, but have placed the wasm tests in their own file. Glad that you made an issue for it, so we can investigate later. Perhaps something to do with disk IO.